src/main/java/org/accounting/system/endpoints/MetricDefinitionEndpoint.java

```diff
@@ -390,7 +393,7 @@ public Response delete(@Parameter(
             schema = @Schema(type = SchemaType.STRING))
                                @PathParam("id") @Valid @NotFoundEntity(repository = MetricDefinitionRepository.class, message = "There is no Metric Definition with the following id:") String id) {
 
-        metricDefinitionService.hasChildren(id);
+        metricDefinitionService.hasChildren(id, "The Metric Definition cannot be deleted. There is a Metric assigned to it.");
```

Comment:
if error message is not configurable or given from another point of the code, better to remove it from being a parameter to the function and add it inside the hasChildren()

Reply:
When updating a metric definition, the same method is used.